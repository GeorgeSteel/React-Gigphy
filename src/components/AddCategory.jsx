import React, { useState } from 'react'
import PropTypes from 'prop-types'

function AddCategory({ setCategories }) {
  const [categorie, setCategorie] = useState('')
  const handleChange = e => setCategorie(e.target.value)
  const handleSubmit = e => {
    e.preventDefault()

    if (categorie === '') return null

    setCategories(prevState => [categorie, ...prevState])
    setCategorie('')
  }

  return (
    <form onSubmit={handleSubmit}>
      <input type='text' value={categorie} onChange={handleChange} />
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
}

export default AddCategory
