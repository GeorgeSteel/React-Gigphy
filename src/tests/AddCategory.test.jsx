import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

import AddCategory from '../components/AddCategory'

describe('AddCategory Component Tests', () => {
  const setCategories = jest.fn()
  let wrapper = shallow(<AddCategory setCategories={setCategories} />)

  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow(<AddCategory setCategories={setCategories} />)
  })

  test('should show the component without issues', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('input value should change', () => {
    const input = wrapper.find('input')
    const value = 'Hello World'

    input.simulate('change', { target: { value } })

    const p = wrapper.find('p')

    expect(p.text().trim()).toBe(value)
  })

  test('should NOT post the data after submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} })

    expect(setCategories).not.toHaveBeenCalled()
  })

  test('should call setCategories and clean the input', () => {
    const input = wrapper.find('input')
    const value = 'Hello World'

    input.simulate('change', { target: { value } })

    wrapper.find('form').simulate('submit', { preventDefault() {} })

    expect(setCategories).toHaveBeenCalled()
    expect(setCategories).toHaveBeenCalledTimes(1)
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function))
    expect(input.prop('value')).toBe('')
  })
})
