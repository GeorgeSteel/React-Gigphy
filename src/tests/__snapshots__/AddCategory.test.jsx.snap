// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory Component Tests should show the component without issues 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
